x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -747,6 +750,9 @@ public void accessGranted(
                 } else if (msg instanceof final SetProfileEnabledRequest setProfileEnabledRequest) {
                     assert SetProfileEnabledAction.NAME.equals(action);
                     securityChangeLogEntryBuilder(requestId).withRequestBody(setProfileEnabledRequest).build();
+                } else if (msg instanceof UpdateApiKeyRequest) {
```

Comment:
We should use `instanceof` pattern matching, i.e.:
```suggestion
                } else if (msg instanceof final UpdateApiKeyRequest updateApiKeyRequest) {
```
There is a meta issue (#81820) that tracks the effort for it as part of rolling out new Java features in the code base. Apparently this class has not been updated yet. But any new code should leverage the new feature.